app/models/manageiq/providers/redhat/infra_manager/event_parser.rb

```diff
@@ -69,18 +69,27 @@ def self.parse_new_target(full_data, message, ems)
 
     {
       :ems_id         => ems.id,
-      :vm             => parse_new_vm(full_data[:vm], message),
+      :vm             => parse_new_vm(full_data[:vm], message, event_type),
       :cluster        => parse_new_cluster(cluster_ref, cluster[:id], cluster_name),
       :resource_pools => parse_new_rp(cluster[:id], cluster_name),
       :folders        => parse_new_dc(full_data[:data_center])
     }
   end
 
-  def self.parse_new_vm(vm, message)
+  def self.parse_new_vm(vm, message, event_type)
     ems_ref = ManageIQ::Providers::Redhat::InfraManager.make_ems_ref(vm[:href])
 
     ManageIQ::Providers::Redhat::InfraManager::RefreshParser.create_vm_hash(
-      ems_ref.include?('/templates/'), ems_ref, vm[:id], message.split(/\s/)[1])
+      ems_ref.include?('/templates/'), ems_ref, vm[:id], parse_target_name(message, event_type))
+  end
+
+  def self.parse_target_name(message, event_type)
+    msg = message.split(/\s/)[1]
+    if %w(NETWORK_ADD_VM_INTERFACE NETWORK_INTERFACE_PLUGGED_INTO_VM).include?(event_type)
+      msg = message.split(/\s/)[8][0...-1]
```

Comment:
You end up setting msg twice for network events, can you do something more like:
```
msg = if NETWORK_EVENTS.include?(event_type)
  message.split(/\s/)[8][0...-1]
else
  message.split(/\s/)[1]
end
```
And in a comment include an example of what the `message` looks like that you're parsing?